product_contract/tests/test_sale_order.py

```diff
@@ -0,0 +1,362 @@
+# Copyright 2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo.exceptions import UserError, ValidationError
+from odoo.fields import Date
+from odoo.tests.common import TransactionCase
+
+
+class TestSaleOrder(TransactionCase):
+    def setUp(self):
+        super(TestSaleOrder, self).setUp()
+        self.product1 = self.env.ref("product.product_product_1")
+        self.product2 = self.env.ref("product.product_product_2")
+        self.sale = self.env.ref("sale.sale_order_2")
+        self.contract_template1 = self.env["contract.template"].create(
+            {"name": "Template 1"}
+        )
+        self.contract_template2 = self.env["contract.template"].create(
+            {
+                "name": "Template 2",
+                "contract_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product2.id,
+                            "name": "Services from #START# to #END#",
+                            "quantity": 1,
+                            "uom_id": self.product2.uom_id.id,
+                            "price_unit": 100,
+                            "discount": 50,
+                            "recurring_rule_type": "yearly",
+                            "recurring_interval": 1,
+                        },
+                    )
+                ],
+            }
+        )
+        self.product1.with_context(force_company=self.sale.company_id.id).write(
```

Comment:
Please use with_company(self.sale.company_id) instead of with_context(force_company=self.company_id.id)